Ee8 schemas

tbitonti opened the pull request:
Parse javaEE8 application.xml and application-client.xml.
Add com.ibm.ws.javaee.platform.v8
Add unit tests for application and application-client.
Add missing javaEE8 unit tests for web-app.

tbitonti commented:
#build

tbitonti commented:
Still have to add cases to the javaEE8 FAT.  See open-liberty/dev/com.ibm.ws.javaee.8.0_fat.

tbitonti commented:
#build

tbitonti commented:
#build

aguibert commented on dev/com.ibm.ws.javaee.platform.v8/build.xml:
```diff
@@ -0,0 +1,13 @@
+<!--
```
this file can be removed, along with the build.xml file from the javaee.platform.v7 project you copied it from (build.xml files are not used in OpenLiberty)

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti commented:
Updated Unit tests:

dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/
  app/AppTest.java
    JavaEEVersion 6.0, 7.0, 8.0
    dtd/schema version 1.4, 5.0, 6.0, 7.0 8.0
  client/AppClientTest.java
    JavaEEVersion 6.0, 7.0, 8.0
    dtd/schema version 1.2, 1.3, 1.4, 5.0, 6.0, 7.0, 8.0
  web/WebAppTestjava
    servlet version 3.0, 3.1, 4.0
    schema version 3.0, 3.1, 4.0
  web/WebFragmentTest.java
    servlet version 3.0, 3.1, 4.0
    schema version 3.0, 3.1, 4.0
- verified successful or failing parse for each combation

Updated FAT tests:

dev/com.ibm.ws.javaee.7.0_fat
    Application cases:
      Simple EAR (v7 application.xml + servlet3.1 web.xml)
      Simple WAR (servlet3.1 web.xml)
    Feature configuration cases:
      webProfile-7.0
      javaee-7.0
- verified application startup and servlet access for each combination

dev/com.ibm.ws.javaee.8.0_fat
    Application cases:
      Simple EAR (v8 application.xml + servlet4 web.xml)
      Simple WAR (servlet4 web.xml)
    Feature configuration cases:
      webProfile-8.0
      javaee-8.0
- verified application startup and servlet access for each combination
